src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerExpediteFinalOrder.java

```diff
@@ -88,6 +88,13 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(CaseDetails<Ca
         log.info("{} about to start callback invoked for Case Id: {}", CASEWORKER_EXPEDITE_FINAL_ORDER, details.getId());
         var caseData = details.getData();
 
+        if (caseData.getConditionalOrder().getGrantedDate() == null) {
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(caseData)
+                .errors(Collections.singletonList("No Conditional Order Granted Date found.  Unable to continue."))
```

Comment:
Nitpick: Might be worth putting the error message in a constant and then we can removed hardcoded strings from these events/tests and will only need to update it once if it changes in the future

Reply:
Good shout - I've done that, prefixing each constant with ERROR_ so they are easy to find with code completion.